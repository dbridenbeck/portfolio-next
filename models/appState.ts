export interface ProjectModel {
  title: string;
  type: string;
  tech: string[];
  projectDetail: string;
  url: string;
  color: string;
  webm: string;
  mp4: string;
  open: boolean;
  thumb: string;
  thumbAlt: string;
}

export default interface AppStateModel {
  currentPage: string;
  pages: [
    { pageName: string; color: string },
    { pageName: string; color: string },
    { pageName: string; color: string }
  ];
  projects: ProjectModel[];
  pageClickedOnce: boolean;
}
