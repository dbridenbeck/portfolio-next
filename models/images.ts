export default interface ImageModel {
  currentPage: string;
  pageClickedOnce: boolean;
}