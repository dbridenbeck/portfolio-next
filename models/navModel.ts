export interface NavProps {
  currentPage: string;
  pages: [
    { pageName: string; color: string },
    { pageName: string; color: string },
    { pageName: string; color: string }
  ];
  changePage: (text: string) => void;
}
