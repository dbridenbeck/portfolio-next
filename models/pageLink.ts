export default interface PageLink {
  onHelloPage: boolean;
}
