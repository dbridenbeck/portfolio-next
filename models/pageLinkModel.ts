export interface PageLinkProps {
  page: { pageName: string; color: string };
  currentPage: string;
  handleChangePage: (text: string) => void;
}
