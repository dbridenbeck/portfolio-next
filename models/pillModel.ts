export interface PillProps {
  children: string;
  setMargin: string;
}
