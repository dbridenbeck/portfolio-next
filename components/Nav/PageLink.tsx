import styled from 'styled-components';
import { PageLinkProps } from "../../models/pageLinkModel";

const PageLinkSpan = styled.span`
  display: block;
  font-size: 1.125em;
  cursor: pointer;
  color: ${({ linkSelected, color }) => linkSelected ? color : "black"};
  :hover {
  color: ${({ color }) => color};
  }
`;

const PageLink: React.FC<PageLinkProps> = ({page, handleChangePage, currentPage}) => {
  const linkSelected = currentPage === page.pageName;
  const updatePage = () => handleChangePage(page.pageName);
  return (
    <>
      <PageLinkSpan 
        color={page.color}
        linkSelected={linkSelected}
        onClick={updatePage}
      >
        {page.pageName}
      </PageLinkSpan>
    </>
  )
};

export default PageLink;