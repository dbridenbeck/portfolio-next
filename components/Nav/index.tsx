import styled from 'styled-components';
import PageLink from "./PageLink";
import { devices } from "../../utils/cssBreakpoints";

const NavContainer = styled.div`
  position: relative;
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  width: 100%;
  margin: 0.5% auto;
  z-index: 99;
  @media ${devices.tabletLandscape} {
    position: fixed;
    justify-content: space-between;
    bottom: 1.333%;
    right: 1.333%;
    width: 20%;
    min-width: 256px;
  }
`;

interface NavProps {
  currentPage: string;
  pages: [
    {pageName: string, color: string}, 
    {pageName: string, color: string}, 
    {pageName: string, color: string} 
  ];
  changePage: (text: string) => void;
}

const Nav: React.FC<NavProps> = ({
  currentPage,
  pages,
  changePage,
}) => (
  <NavContainer >
    {pages.map(page => (
      <PageLink 
        page={page}
        key={page.pageName}
        currentPage={currentPage}
        handleChangePage={changePage}
      />
    ))}
  </NavContainer>
);

export default Nav;
