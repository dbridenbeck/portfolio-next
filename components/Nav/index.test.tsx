import React from "react";
import { render } from "@testing-library/react";
import { NavProps } from "../../models/navModel";
import Nav from "./index";

describe("Nav", () => {
  const requiredProps: NavProps = {
    pages: [
      {
        pageName: "portfolio",
        color: "#427ce6",
      },
      {
        pageName: "skills",
        color: "#6f59e9",
      },
      {
        pageName: "about",
        color: "#b0292e",
      },
    ],
    currentPage: "about",
    changePage: jest.fn(),
  };

  it("should render without errors", () => {
    const { container } = render(<Nav {...requiredProps} />);
    expect(container).toMatchSnapshot();
  });
});
