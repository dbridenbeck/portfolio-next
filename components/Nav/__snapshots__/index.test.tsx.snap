// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render without errors 1`] = `
<div>
  <div
    class="Nav__NavContainer-sc-1ayp5y4-0 limDtW"
  >
    <span
      class="PageLink__PageLinkSpan-x2cztj-0 fLJgUU"
      color="#427ce6"
    >
      portfolio
    </span>
    <span
      class="PageLink__PageLinkSpan-x2cztj-0 kGmCBl"
      color="#6f59e9"
    >
      skills
    </span>
    <span
      class="PageLink__PageLinkSpan-x2cztj-0 xWTRx"
      color="#b0292e"
    >
      about
    </span>
  </div>
</div>
`;
